### PR TITLE
Bug/2102 label owasp scans fail

### LINKED ISSUE #2102: Owasp scans deployed from a label fail
## Description

OWASP scans triggered by a label are failing for two unrelated reasons. 

## Action Taken

1. Changes made to frontend deployment cause build step to fail, resulting in this error:
https://app.circleci.com/pipelines/github/raft-tech/TANF-app/10276/workflows/6923096a-aefd-4660-ba08-63a050ab6787/jobs/26227/parallel-runs/0/steps/0-103
This can be remedied by updating the build command in the frontend docker script like so:
https://github.com/raft-tech/TANF-app/commit/5a88aaf6e0b3d0b99079b081af2d675618b76af6
This was done in testing and resulted in a new error that seems more prevalent.

2. After resolving the script error there should be an error in the scan step:
https://app.circleci.com/pipelines/github/raft-tech/TANF-app/10221/workflows/5f3ef60a-66c0-442e-9d87-77f6cf2fd953/jobs/26077/parallel-runs/0/steps/0-104
It appears to be a pip dependency versioning error.

## What I expected to see

When a branch is given the `QASP Review` label owasp should run without error.

## What I did see

When adding the Qasp Review label the scan fails.

## Other Helpful Information
See above links.

### PR DESCRIPTION
**Pull request by n0remac**
## Summary of Changes
_Provide a brief summary of changes_
Pull request closes #2102 

## How to Test
_List the steps to test the PR_
Make sure all tests have passed. 
Add the QASP Review label.
Labeled owasp scan should run without issue.

Screen Shot 2022-09-29 at 1 20 36 PM

## Deliverables
_More details on how deliverables herein are assessed included [here](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 2: Tested Code](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
  + [ ] If this PR introduces backend code changes, are they meaningfully tested?
  + [ ] If this PR introduces frontend code changes, are they meaningfully tested?
+ Are code coverage minimums met?
  + [ ] Frontend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)
  + [ ] Backend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)

### [Deliverable 3: Properly Styled Code](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)

+ [x] Are backend code style checks passing on CircleCI?
+ [x] Are frontend code style checks passing on CircleCI?
+ [x] Are code maintainability principles being followed?

### [Deliverable 4: Accessible](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)

+ [ ] Does this PR complete the epic? 
+ [ ] Are links included to any other gov-approved PRs associated with epic?
+ [ ] Does PR include documentation for Raft's a11y review? 
+ [ ] Did automated and manual testing with `iamjolly` and `ttran-hub` using Accessibility Insights reveal any errors introduced in this PR?

### [Deliverable 5: Deployed](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)

+ [x] Was the code successfully deployed via automated CircleCI process to development on Cloud.gov?

### [Deliverable 6: Documented](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)

+ [x] Does this PR provide background for why coding decisions were made?
+ [ ] If this PR introduces backend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces frontend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces dependencies, are their licenses documented?
+ [ ] Can reviewer explain and take ownership of these elements presented in this code review?

### [Deliverable 7: Secure](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)

+ [x] Does the OWASP Scan pass on CircleCI?
+ [ ] Do manual code review and manual testing detect any new security issues?
+ [ ] If new issues detected, is investigation and/or remediation plan documented? 

### [Deliverable 8: User Research](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
+ [ ] the purpose of the research
+ [ ] methods used to conduct the research 
+ [ ] who participated in the research
+ [ ] what was tested and how
+ [ ] impact of research on TDP
+ [ ] (_if applicable_) final design mockups produced for TDP development

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -475,7 +475,7 @@ jobs:
             docker-compose run --rm zaproxy bash -c \
               "PATH=$PATH:/home/zap/.local/bin &&
                pip install wait-for-it &&
-               wait-for-it --service http://web:3000 \
+               wait-for-it --service http://tdp-frontend/ \
```
Comment (n0remac) [inline .circleci/config.yml:478]: Using the right url here seemed to be one of the important steps. The wait for it command would time out if the port was set.

File: tdrs-frontend/Dockerfile [hunk 2]
```diff
@@ -27,7 +27,7 @@ EXPOSE 80
 # ---
 # Stage 1: Create a production build
 FROM localdev AS build
-RUN npm run build
+RUN npm run build:development
```
Comment (n0remac) [inline tdrs-frontend/Dockerfile:30]: This is important to set the env variables.

File: tdrs-frontend/nginx/nginx.conf [hunk 3]
```diff
@@ -12,8 +12,4 @@ server {
     add_header Content-Security-Policy "default-src 'self'; *.acf.hhs.gov; script-src 'self'; style-src 'self' 'unsafe-inline'; img-src 'self' data:; font-src 'self'; connect-src 'self'; *.acf.hhs.gov manifest-src 'self'; object-src 'none'; frame-ancestors 'none'; form-action 'none';";
     add_header Access-Control-Allow-Origin "https://tanfdata.acf.hhs.gov";
 
-     location / {
-         proxy_pass http://$host;
-         proxy_set_header Host $host;
-     }
```
Comment (n0remac) [inline tdrs-frontend/nginx/nginx.conf:18]: This part is what ultimately broke the scans. It seems nginx was attempting to proxy traffic to somewhere that does not exist.

Comment (n0remac) [reply tdrs-frontend/nginx/nginx.conf:18]: I deployed this to dev raft and it still works. The `add_header Access-Control-Allow-Origin "https://tanfdata.acf.hhs.gov";` makes it seem like this nginx conf is for production, but on closer inspection that does not seem to be the case. The buildpack.nginx.conf seems to be used for that. This line is likely not needed here, thus this change should not have an effect on production.

Still we should test this commit on prod or be ready to revert it if it does break something.

Comment (raftmsohani) [reply tdrs-frontend/nginx/nginx.conf:18]: For local dev, Nginx is serving frontend build package. @n0remac  you are right, for production, the Nginx conf is in buildpack.nginx.conf. I think at some point, we needed this line for difference between dev and circleci, but not needed anymore.

**Review by raftmsohani — APPROVED** [review]
LGTM!

**Review by andrew-jameson — APPROVED** [review]

**Review by jtimpe — APPROVED** [review]

**Review by ADPennington — APPROVED** [review]
thank you @n0remac 🚀

**Pull request merged**